app/controllers/ems_cloud_controller.rb

```diff
@@ -149,7 +149,18 @@ def ems_cloud_form_fields
       zone = @ems.my_zone
     end
 
-    amqp_userid = @ems.has_authentication_type?(:amqp) ? @ems.authentication_userid(:amqp).to_s : ""
+    amqp_userid = ""
+    amqp_hostname =  ""
+    amqp_port =  ""
+
+    if @ems.connections.amqp.try(:endpoint)
```

Comment:
On a minor side, I prefer [try!](http://api.rubyonrails.org/v4.2.5/classes/Object.html#method-i-try-21) if _try_ ing is unavoidable.